astropy/modeling/tests/test_fitters.py

```diff
@@ -468,6 +468,25 @@ def test_LSQ_SLSQP_with_constraints(self):
         assert_allclose(model.parameters, slsqp_model.parameters,
                         rtol=10 ** (-4))
 
+    @pytest.mark.filterwarnings(r'ignore:Model is linear in parameters; consider using linear fitting methods.')
+    def test_LevMar_with_weights(self):
+        """
+        Tests that issue #11581 has been solved.
+        """
+        norder = 2
+        tx = np.arange(10)
+        ty = np.arange(10)
+        tw = np.arange(10)
+        fitter1 = LevMarLSQFitter()
+        fitter2 = LinearLSQFitter()
+        bkg_model = models.Polynomial1D(norder)
+
+        tf1 = fitter1(bkg_model, tx, ty, weights=tw)
```

Comment:
Instead of filtering warnings via `@pytest.mark.filterwarnings` you could actually make sure just the `LevMarLSQFitter` raises the expected warning:

```suggestion
        with pytest.warns(AstropyUserWarning, match="Model is linear in parameters"):
            tf1 = fitter1(bkg_model, tx, ty, weights=tw)
```